File: src/test/java/com/github/xumeng/java/problems/easy/problem0338/Solution0338Test.java
package com.github.xumeng.java.problems.easy.problem0338;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class Solution0338Test {

    private final Solution0338 solution = new Solution0338();


    @Test
    public void test1() {
        int n = 0;
        int[] result = solution.countBits(n);
        int[] expected = {0};
        assertArrayEquals(expected, result);
    }

    @Test
    public void test2() {
        int n = 1;
        int[] result = solution.countBits(n);
        int[] expected = {0, 1};
        assertArrayEquals(expected, result);
    }

    @Test
    public void test3() {
        int n = 2;
        int[] result = solution.countBits(n);
        int[] expected = {0, 1, 1};
        assertArrayEquals(expected, result);
    }

    @Test
    public void test4() {
        int n = 5;
        int[] result = solution.countBits(n);
        int[] expected = {0, 1, 1, 2, 1, 2};
        assertArrayEquals(expected, result);
    }

}

File: src/test/kotlin/com/github/xumeng/kotlin/problems/easy/problem0338/Solution0338Test.kt
package com.github.xumeng.kotlin.problems.easy.problem0338

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class Solution0338Test {

    private val solution = Solution0338()

    @Test
    fun test1() {
        val n = 0
        val result = solution.countBits(n)
        val expected = intArrayOf(0)
        assertArrayEquals(expected, result)
    }

    @Test
    fun test2() {
        val n = 1
        val result = solution.countBits(n)
        val expected = intArrayOf(0, 1)
        assertArrayEquals(expected, result)
    }

    @Test
    fun test3() {
        val n = 2
        val result = solution.countBits(n)
        val expected = intArrayOf(0, 1, 1)
        assertArrayEquals(expected, result)
    }

    @Test
    fun test4() {
        val n = 5
        val result = solution.countBits(n)
        val expected = intArrayOf(0, 1, 1, 2, 1, 2)
        assertArrayEquals(expected, result)
    }

}